bin/rad-diff

```diff
@@ -0,0 +1,350 @@
+    (match (process! "git" ["checkout" "-b" (string-append "diff/" (show diff-no))] "")
+      :ok (apply-patch! (lookup :patch diff))
+      _   (put-str! (string-append "Command 'git checkout -b diff/" (show diff-no) "' failed")))))
+
+(def push-diff!
+  "Merges the DIFF to master, and pushes the changes."
+  (fn [chain-ref diff-no]
+    (def diff (verify-diff-number diff-no chain-ref))
+    (process! "git" ["checkout" "master"] "")
+    (match (apply-patch! (lookup :patch diff))
+      :ok (do
+            (put-str! "ok!")
+            (process! "git" ["push"] ""))
+      _   (do
+            (put-str! "Resolve any conflicts and then run 'git push'")
+            (exit! 1)))))
+
+;; create a new diff
+
+(def new-diff!
+  (fn [chain commit]
+    (verify-commit commit)
+    (def patch (get-patch! commit))
+    (def author-name (get-author-name! commit))
+    (def author-email (get-author-name! commit))
+    (def short-hash (get-short-hash! commit))
+    (def title (get-title! commit))
+    (def description (get-description! commit))
+    (def diff {:short-hash    short-hash
+               :title         title
+               :description   description
+               :author-name   author-name
+               :author-email  author-email
+               :patch         patch})
+    (simple-create-diff! (make-chain-ref chain) diff)
+    (put-str! (string-append "Proposing " short-hash " to " chain))))
+
+;; showing diffs
+
+(def pretty-state
+  (fn [state]
+    (match state
+      :pending    "pending"
+      :retracted  "retracted"
+      :accepted   "accepted"
+      :rejected   "rejected")))
+
+(def pretty-headline
+  (fn [i]
+    (match i
+      '{:number n
+        :title tit
+        :state s
+        :author-name an
+        :short-hash sh}
+      (string-append
+        "(" (pretty-state s) ") [" an "] " sh " - " tit " | DIFF " (show n)))))
+
+(def pretty-comment
+ (fn [c]
+   (match c
+     '{:created-at c-a
+       :body b
+       :author-name a-n}
+     (string-append "### " a-n " [" c-a "]\n\n" b "\n"))))
+
+(def pretty-diff
+  (fn [i]
+    (match i
+      '{:patch p
+        :created-at c-a
+        :comments cs}
+      (string-append
+        (pretty-headline i) "\n\n"
+        "created at " c-a "\n\n"
+        p "\n\n"
+        "Comments\n"
+        "--------\n\n"
+        (string/unlines (map pretty-comment cs))))))
+
+(def comment-divider "----------------------------------------")
+
+(def add-comment-md
+  (fn [x]
+    (string-append
+     "\n" comment-divider "\n"
+     "---Add comment above the line.\n\n"
+     x)))
+
+(def list-items
+  "Lists the recent items and uses an editor to add a relevant comment."
+  (fn [chain options]
+    (def chain-ref (make-chain-ref chain))
+
+    (def mk-key
+      (fn [v]
+        (match v
+               (/as 'a {:activity :new-diff
+                        :diff     'diff})
+               [(string-append "[new/changed] " (pretty-headline diff)) a]
```

Comment:
Can you indent this differently, so it's clearer what the match is, and what the corresponding expression?